### PR TITLE
XFS support

### LINKED ISSUE #31: Xfs not working on centos machines
Hi,

xfstools are not installed on centos like machines, thus failing xfs mountpoint

I did a huge PR adding molecule tests and fixing that case.

### PR DESCRIPTION
**Pull request by Tcharl**
fix #31

### CODE REVIEW

**Review by markgoddard — COMMENTED** [review]
I think some unnecessary files have been committed:

`.idea/`
`molecule/default/tests/__pycache__/`

File: .github/workflows/molecule.yml [hunk 1]
```diff
@@ -0,0 +1,100 @@
+---
```
Comment (markgoddard) [inline .github/workflows/molecule.yml:1]: I've been meaning to look into Github actions. Will these be run in this PR? Where can I see the results? I only see travis.

File: .github/workflows/molecule.yml [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+on:
+  push:
+  pull_request:
+    branches:
+      - master
+      - tags/*
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # This workflow contains a single job called "build"
```
Comment (markgoddard) [inline .github/workflows/molecule.yml]: Comment incorrect

File: .github/workflows/molecule.yml [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+          sudo pip3 install docker molecule testinfra yamllint ansible-lint flake8 molecule-vagrant
+
+      - name: molecule lint
+        run: |
+          molecule lint
+
+  molecule:
+
+    runs-on: ubuntu-latest
+    if: "false"
```
Comment (markgoddard) [inline .github/workflows/molecule.yml:46]: So it is disabled?

File: tasks/debian.yml [hunk 4]
```diff
@@ -1,22 +1,40 @@
...
-  apt:
+  package:
     name:
       - lvm2
       - scsitools
-      - xfsprogs
     state: "present"
   become: true
 
+- name: centos | install xfs tools
```
Comment (markgoddard) [inline tasks/debian.yml]: s/centos/debian/

File: tasks/create_fs.yml [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+    lv.filesystem is defined and
+    lv.filesystem == 'xfs'
+    )
+    )
+
+- name: "debug mount"
+  loop: "{{ vg.lvnames }}"
+  debug: var=lv
+  loop_control:
+    loop_var: lv
```
Comment (markgoddard) [inline tasks/create_fs.yml]: Do we want these debug statements?

File: tasks/create_fs.yml [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+  become: true
+  loop: "{{ vg.lvnames }}"
+  loop_control:
+    loop_var: lv
+  when: >
+    (lv is defined and
+    lv != 'None') and
+    (lv.create is defined and
+    not lv.create and
+    lv.filesystem != "swap")
```
Comment (markgoddard) [inline tasks/create_fs.yml:17]: If you pull this task out of this file, you could add the following condition to the include, and remove it from the tasks in this file:

```
when: vg.create is defined and vg.create
```

Same applies to create_lv.yml

File: tasks/debian.yml [hunk 7]
```diff
@@ -1,22 +1,40 @@
...
+  package:
+    name: "xfsprogs"
+    state: "present"
+  become: yes
+  loop: "{{ lvm_groups|subelements('lvnames') }}"
```
Comment (markgoddard) [inline tasks/debian.yml:15]: It's a shame that this could run multiple times. It's not that big a deal though.

File: tasks/main.yml [hunk 8]
```diff
@@ -8,5 +8,5 @@
 
 - include: manage_lvm.yml
```
Comment (markgoddard) [inline tasks/main.yml]: include_tasks is the modern way (since you have switched to loop)

**Review by markgoddard — CHANGES_REQUESTED** [review]
Thanks. The addition of molecule tests is nice.

As a general comment I would say that this patch was quite difficult to review because it included too many things. In particular, when you mix a refactor with a code change, it's very hard to see what has changed vs. what has just moved.

File: .github/workflows/molecule.yml (hunk 1 above)
Comment (Tcharl) [reply .github/workflows/molecule.yml:1]: You'll see github actions when you'll accept the request and go to your repo settings to activate actions.
I didn't manage to make it work with vagrant-libvirt but it's only temporary, when I'll have the time

File: .github/workflows/molecule.yml (hunk 3 above)
Comment (Tcharl) [reply .github/workflows/molecule.yml:46]: Yes it is, doesn't work for now (VM in docker) but I'll try to make it work in a little days

File: .github/workflows/molecule.yml (hunk 1 above)
Comment (Tcharl) [reply .github/workflows/molecule.yml:1]: Only lint is activated for now

File: tasks/debian.yml (hunk 7 above)
Comment (Tcharl) [reply tasks/debian.yml:15]: Sure, do you know how to assess that there's a match in a list?

File: tasks/create_fs.yml (hunk 6 above)
Comment (Tcharl) [reply tasks/create_fs.yml:17]: I just took the original guards: I don't even know why we should check that!
To me, lv.create check should be sufficient to check the fs creation?

File: .github/workflows/molecule.yml (hunk 3 above)
Comment (Tcharl) [reply .github/workflows/molecule.yml:46]: Note that with a vagrant and virtualbox installation, you can test it locally :-)

File: tasks/create_fs.yml (hunk 6 above)
Comment (markgoddard) [reply tasks/create_fs.yml:17]: I think the guards are right - we may want to delete a VG by setting vg.create = false. Same for an LV. We also may want to create a VG and LV, but not a file system, or create a file system but not mount it.

File: tasks/debian.yml (hunk 7 above)
Comment (markgoddard) [reply tasks/debian.yml:15]: Two options I can think of:
1. add a task that uses this loop with set_fact, then install the package based on the fact.
2. some complicated unreadable jinja expression using a few `selectattr`s.

The former is probably preferable :)

Comment (Tcharl) [reply tasks/debian.yml:15]: Not a fan of set fact: to me you should use it to add a machine attribute (os kind or hardware), not something related to installed software.
May be with a register somehow

**Comment by Tcharl** [comment]
Anyway, do you think it's good to go? Remaining remarks looks nice to have and I don't think I'll have so much time to spend on it: nice to have are at the bottom of my totally overloaded backlog :-p

**Review by markgoddard — APPROVED** [review]
LGTM, thanks

**Comment by mrlesmithjr** [comment]
@markgoddard Thanks so much for taking the time to review this. @Tcharl Thanks for taking your time as well contributing. I'll get this merged now.

**Pull request merged**